### PR TITLE
19 implement spare tiles

### PR DESCRIPTION
**Pull request by zientist**

### CODE REVIEW

File: src/alcaramba/Drawable/Town/Town.tscn [hunk 1]
```diff
@@ -43,8 +48,30 @@ margin_right = 185.0
...
 script = ExtResource( 2 )
 
+[node name="ContainerRight" type="VBoxContainer" parent="."]
+material = SubResource( 3 )
+margin_left = 695.0
+margin_top = 314.0
+margin_right = 1050.0
+margin_bottom = 912.0
+
+[node name="CurrentTile" type="VBoxContainer" parent="ContainerRight"]
```
Comment (Structed) [inline src/alcaramba/Drawable/Town/Town.tscn:58]: Seems unused. Remove if it is.

File: src/alcaramba/Drawable/Town/Town.tscn [hunk 2]
```diff
@@ -43,8 +48,30 @@ margin_right = 185.0
...
+margin_right = 355.0
+margin_bottom = 50.0
+rect_min_size = Vector2( 0, 50 )
+
+[node name="SpareTiles" type="VBoxContainer" parent="ContainerRight"]
```
Comment (Structed) [inline src/alcaramba/Drawable/Town/Town.tscn:63]: Needs to be size-capped to the end of the screen and needs a scrollbar, so you can access "off-screen" tiles

File: src/alcaramba/Drawable/Town/Town.tscn [hunk 3]
```diff
@@ -43,8 +48,30 @@ margin_right = 185.0
...
+unique_name_in_owner = true
+margin_top = 54.0
+margin_right = 355.0
+margin_bottom = 104.0
+rect_min_size = Vector2( 50, 50 )
+script = ExtResource( 4 )
+
```
Comment (Structed) [inline src/alcaramba/Drawable/Town/Town.tscn:70]: Can you please add the functionality to directly place a tile from the Market to the spare yard?

**Review by Structed — CHANGES_REQUESTED** [review]
Looks good, but there's a bit of missing functionality and some clean-up missing.

File: src/alcaramba/Drawable/Game/Game.gd [hunk 4]
```diff
@@ -9,23 +9,30 @@ onready var _market := get_node("%Market")
...
 
-
+# if tile is placed add it to town stack and remove selection
 func _on_tile_placed(tile: TileCard, position: Vector2):
 	Global.active_player.add_town_tile(tile, position)
 	_selected_tile.clear()
+
+# get to state with no selected tile, remove highlight if there was a previously selected tile
+func _deselect_tile():
+	if _selected_tile != null:
```
[the remark this answers is not in the document]
Comment (Structed) [reply src/alcaramba/Drawable/Game/Game.gd:36]: After discussion, this is better. This leads to actually always deselecting the tile

File: src/alcaramba/Drawable/Town/Town.tscn (hunk 2 above)
Comment (zientist) [reply src/alcaramba/Drawable/Town/Town.tscn:63]: Just for prototyping. UX and design has to done in future task

File: src/alcaramba/Drawable/Town/Town.tscn (hunk 3 above)
Comment (zientist) [reply src/alcaramba/Drawable/Town/Town.tscn:70]: out of scope

**Review by Structed — APPROVED** [review]

**Pull request merged**